--- /home/maxrdlf95/cpu-tegra3dmore.c	2013-05-28 17:38:49.555699408 -0430
+++ /home/maxrdlf95/cpu-tegra3faux.c	2013-05-28 17:39:01.947699255 -0430
@@ -41,7 +41,7 @@
 #define INITIAL_STATE		TEGRA_HP_DISABLED
 #define UP2G0_DELAY_MS		70
 #define UP2Gn_DELAY_MS		100
-#define DOWN_DELAY_MS		2000
+#define DOWN_DELAY_MS		500
 
 static struct mutex *tegra3_cpu_lock;
 
@@ -66,7 +66,7 @@
 static int mp_overhead = 10;
 module_param(mp_overhead, int, 0644);
 
-static int balance_level = 70;
+static int balance_level = 60;
 module_param(balance_level, int, 0644);
 
 static struct clk *cpu_clk;
@@ -151,12 +151,9 @@
 
 	if (ret == 0) {
 		if ((hp_state == TEGRA_HP_DISABLED) &&
-				(old_state != TEGRA_HP_DISABLED)) {
-					mutex_unlock(tegra3_cpu_lock);
-					cancel_delayed_work_sync(&hotplug_work);
-					mutex_lock(tegra3_cpu_lock);
-					pr_info("Tegra auto-hotplug disabled\n");
-		} else if (hp_state != TEGRA_HP_DISABLED) {
+		    (old_state != TEGRA_HP_DISABLED))
+			pr_info("Tegra auto-hotplug disabled\n");
+		else if (hp_state != TEGRA_HP_DISABLED) {
 			if (old_state == TEGRA_HP_DISABLED) {
 				pr_info("Tegra auto-hotplug enabled\n");
 				hp_init_stats();
@@ -194,7 +191,7 @@
 #define NR_FSHIFT	2
 static unsigned int nr_run_thresholds[] = {
 /*      1,  2,  3,  4 - on-line cpus target */
-	5,  9, 13, UINT_MAX /* avg run threads * 4 (e.g., 9 = 2.25 threads) */
+	5,  9, 10, UINT_MAX /* avg run threads * 4 (e.g., 9 = 2.25 threads) */
 };
 static unsigned int nr_run_hysteresis = 2;	/* 0.5 thread */
 static unsigned int nr_run_last;
@@ -347,13 +344,13 @@
 
 	if (cpu < nr_cpu_ids) {
 		if (up){
-			pr_debug("cpu_up(%u)+\n",cpu);
+			printk("cpu_up(%u)+\n",cpu);
 			cpu_up(cpu);
-			pr_debug("cpu_up(%u)-\n",cpu);
+			printk("cpu_up(%u)-\n",cpu);
 		}else{
-			pr_debug("cpu_down(%u)+\n",cpu);
+			printk("cpu_down(%u)+\n",cpu);
 			cpu_down(cpu);
-			pr_debug("cpu_down(%u)-\n",cpu);
+			printk("cpu_down(%u)-\n",cpu);
 		}
 	}
 }
@@ -438,7 +435,7 @@
 			hp_state = TEGRA_HP_DOWN;
 			queue_delayed_work(
 #ifdef CONFIG_TEGRA_RUNNABLE_THREAD
-				hotplug_wq, &hotplug_work, up2gn_delay);
+				hotplug_wq, &hotplug_work, up_delay);
 #else
 				hotplug_wq, &hotplug_work, down_delay);
 #endif
@@ -605,11 +602,7 @@
 	pm_qos_add_request(&min_cpu_req, PM_QOS_MIN_ONLINE_CPUS,
 			   PM_QOS_DEFAULT_VALUE);
 	pm_qos_add_request(&max_cpu_req, PM_QOS_MAX_ONLINE_CPUS,
-#ifdef DEFAULT_DUAL_CORE
-			   (s32)2);
-#else
 			   PM_QOS_DEFAULT_VALUE);
-#endif
 
 	if (!debugfs_create_file(
 		"min_cpus", S_IRUGO, hp_debugfs_root, NULL, &min_cpus_fops))
